lib/fs.js

```diff
@@ -293,7 +303,12 @@ ReadFileContext.prototype.close = function(err) {
   req.oncomplete = readFileAfterClose;
   req.context = this;
   this.err = err;
-  binding.close(this.fd, req);
+
+  if (this.isUserFd) {
+    req.oncomplete.call(req, null);
```

Comment:
This needs to be clearly documented. Possibly with big bold text. I can see users thinking they're handing off the fd.